docs/gradle-development.rst

```diff
@@ -90,3 +88,6 @@ This will trigger two GitHub Actions builds and the generated artifacts will be
 to the OME Artifactory. All builds from the `master` branch are expected to be snapshots
 and uploaded to the `ome.snapshots` repository. All tag builds are expected to be full
 releases and uploaded to the `ome.releases` repository.
+
+On GitHub, create a release from the `vx.y.z` tag e.g. using https://github.com/ome/omero-server/releases/new
```

Comment:
I think we could automate that via the Github action later
but for now it makes sense